sdk/src/main/java/com/horizen/account/state/EvmMessageProcessor.java

```diff
@@ -24,26 +27,36 @@ public boolean canProcess(Message msg, BaseAccountStateView view) {
     }
 
     @Override
-    public ExecutionResult process(Message msg, BaseAccountStateView view) {
-        try {
-            // TODO: this will only process legacy transactions correctly as PriorityFee and FeeCap are ignored
-            //  (and the baseFee is set to zero internally in `LibEvm.evmApply`)
-            var result = Evm.Apply(
-                    view.getStateDbHandle(),
-                    msg.getFrom().address(),
-                    msg.getTo() == null ? null : msg.getTo().address(),
-                    msg.getValue(),
-                    msg.getData(),
-                    msg.getGasLimit(),
-                    msg.getGasPrice()
-            );
-            if (result.evmError.isEmpty()) {
-                return new ExecutionSucceeded(result.usedGas, result.returnData);
+    public byte[] process(Message msg, BaseAccountStateView view) throws ExecutionFailedException {
+        // prepare context
+        var context = new EvmContext();
+        context.baseFee = view.getBaseFee();
+        context.blockNumber = BigInteger.valueOf(view.getHeight());
+        // total gas limit supplied in the transaction, only used by the GASLIMIT opcode
+        // current gas in the pool cannot be used because intrinsic gas was already removed here
+        context.gasLimit = msg.getGasLimit();
```

Comment:
The only purpose of the gas limit in the context is to be available in case a smart contract uses the op-code `GASLIMIT`.

And yes, what you said is how I intended it to work. Now that you mentioned it I double checked and it turns out that op-code is supposed to return the gas limit of the block instead, not of the current transaction.

I'm gonna fix that :see_no_evil:

Reply:
fixed